src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -902,7 +906,7 @@ else if ( CompilerConfiguration.CompilerReuseStrategy.ReuseSame.getStrategy().eq
                 {
                     String cause = idk ? "idk" : ( dependencyChanged ? "dependency"
                             : ( sourceChanged ? "source" : "input tree" ) );
-                    getLog().info( "Changes detected - recompiling the module! :" + cause );
+                    getLog().info( "Changes detected! - recompiling the module due to: " + cause );
```

Comment:
If this message is going to change - could we drop exclamation point completely?